agent/netif.go

```diff
@@ -45,7 +45,7 @@ func (netif *NetIf) UnmarshalJSON(data []byte) error {
 	m := make(map[string]string)
```

Comment:
Can't attach comment to line 42 (the docstring), so I'll use this line: Yes, we can see that it implements the unmarshal interface. But why is that needed? It's a little unusual to have a low-level function deal with data in JSON like that, so maybe a few words about the why?